geopandas/geodataframe.py

```diff
@@ -463,7 +526,7 @@ def rename_geometry(self, col, inplace=False):
             self.set_geometry(col, inplace=inplace)
 
     @property
-    def active_geometry_name(self):
+    def active_geometry_name(self) -> str:
```

Comment:
Mostly string but not necessarily. We can easily have a geometry column indexed by an integer. Isn't it any hashable object, technically?

Reply:
Are you talking to me, because I have no idea 🤣

Reply:
I suppose to @m-richards at this point :)

Reply:
Yep, good catch, I probably just translated the docstring type on autopilot. Pandas is not very precise about the type of df.index.name and I think this case is equivalent to that.